app/javascript/mastodon/selectors/index.js

```diff
@@ -8,57 +8,98 @@ import { getFilters } from './filters';
 export { makeGetAccount } from "./accounts";
 export { getStatusList } from "./statuses";
 
-export const makeGetStatus = () => {
-  return createSelector(
-    [
-      (state, { id }) => state.getIn(['statuses', id]),
-      (state, { id }) => state.getIn(['statuses', state.getIn(['statuses', id, 'reblog'])]),
-      (state, { id }) => state.getIn(['accounts', state.getIn(['statuses', id, 'account'])]),
-      (state, { id }) => state.getIn(['accounts', state.getIn(['statuses', state.getIn(['statuses', id, 'reblog']), 'account'])]),
-      getFilters,
-      (_, { contextType }) => ['detailed', 'bookmarks', 'favourites'].includes(contextType),
-    ],
-
-    (statusBase, statusReblog, accountBase, accountReblog, filters, warnInsteadOfHide) => {
-      if (!statusBase || statusBase.get('isLoading')) {
-        return null;
-      }
+const getStatusInputSelectors = [
+  (state, { id }) => state.getIn(['statuses', id]),
+  (state, { id }) => state.getIn(['statuses', state.getIn(['statuses', id, 'reblog'])]),
+  (state, { id }) => state.getIn(['accounts', state.getIn(['statuses', id, 'account'])]),
+  (state, { id }) => state.getIn(['accounts', state.getIn(['statuses', state.getIn(['statuses', id, 'reblog']), 'account'])]),
+  getFilters,
+  (_, { contextType }) => ['detailed', 'bookmarks', 'favourites'].includes(contextType),
+];
 
-      if (statusReblog) {
-        statusReblog = statusReblog.set('account', accountReblog);
-      } else {
-        statusReblog = null;
-      }
+function getStatusResultFunction(
```

Comment:
The only change in this function is that it now returns an object of the shape below depending on the case, rather than just `Status | null`.

```ts
{
  status: Status | null;
  loadingState: 'not-found' | 'loading' | 'filtered' | 'complete';
}
```